### ISSUE TITLE
navigation-tree without link to page.md 

### ISSUE THREAD

**Issue by lerchchr** [datenstrom/yellow#462]
Suppose this structure for the content of the website:
```
├── content
    ├── foo
        ├── foo.md
    ├── bar
        ├── bar.md
        ├── bar1.md
        ├── bar2.md
    └── bas
        └── bas.md
```
Clicking on `foo` refers to `http://website/`, hovering shows also `http://website/`.
Clicking on `bas` refers to `http://website/bas`, hovering shows also `http://website/bas`.

Clicking on bar refers to `http://website/bar`, hovering shows also `http://website/bar`. However, `bar.md` has no meaningful content.
The desired behaviour is that hovering `bar` only shows the navigation-tree with `bar1` and `bar2` and no link to `bar`.
Is there any chance to achieve this (easily)?

**Comment by markseu** [comment]
Hello lerchchr, I am not sure what kind of problem you want to solve. Is this about understanding content structure or making a custom navigation layout? Can you give an describe a little bit more?

**Comment by lerchchr** [comment]
This is about making a custom navigation layout:
Within `content` every folder with a prefix is shown in the navigation. However, every folder must contain a file called `page.md` (or with the name of the folder), which itself renders to a html-page.
For my website, it makes no sense to create a html-page with just the name of the folder. Instead, I want the name of the folder to be shown in the navigation (but it should not be clickable, as there is no html-page linked to it), hovering should instead show a navigation-tree with the remaining html-pages in this content folder (defined by the additional files in this content folder).
Example: https://www.auferstehungskirchengemeinde.de/, hovering "Angebote"

**Comment by markseu** [comment]
Thanks for the explanation. I am sorry to say, the default navigation-tree does not support this feature. You can make your own navigation or modify the existing one. Have a look at the API, for example check if an element has children before generating an HTML link.

**Comment by BsNoSi** [comment]
Hi Lerchchr,

I took a short look on your referenced website and read your desire carefully. I want to point out something you possibly might should think about: Is it »good practice« to offer a folder content only without description of it's contents?

I for myself love short introductions what I can expect in an area of a website. I find it irritating, having no explanation what to expect by opening "Predigten → Anhören" and what is the idea behind. A good sermon also beginns with an introduction instead of coming straight to the point, doesn't it?

Another point you may think about: opening "Angebote" on a mobile device covers the navigaiton – with no "point of return". I have to refresh the page or take a detour over "Gottestdienste" or "Aktuelles" to close the pop up. I respect hat you may have a different point of view but I feel this is not really handy or user friendly.

Just my thoughts. No offense.

**Comment by markseu** [comment]
![Screen Shot Auferstehungskirchengemeinde](https://user-images.githubusercontent.com/2429165/68991548-2e023280-0860-11ea-9258-85410fc3ba5c.png)

I am surprised that not more people make custom navigations, like in the screenshot above.

There is no rule that a navigation has to represent the physical folder structure of the website. I can image a navigation that guides users to the most interesting starting points. Now the whole website can become part of the navigation experience. Navigation is overdone on many sites. Especially on a mobile phone, I like the idea of swiping through the page and see more content unfolding on the whole page. Rather than clicking on a hamburger menu or the navigation part of a page. You scrolled down while you are reading this page and know what I mean. User research by Jakob Nielsen confirms this, users look straight at the content and ignore the navigation areas. [[1]](https://www.nngroup.com/articles/is-navigation-useful/) [[2]](https://www.nngroup.com/articles/top-ten-guidelines-for-homepage-usability/)

For a long time, I have been playing with the idea of removing navigation-tree layout from the default installation. Instead let the API come with different examples, let themes come with different ready-to-use implementations. This might lead to two things, more developers experimenting with custom navigations, more people having access to themes with custom layouts.

TLDR: I hope to see more custom navigations. Help users find what they need.

**Comment by lerchchr** [comment]
You both have valid points, thank you. I am still undecided.
Meanwhile I have been fiddling around with of-else-statements and their positions within li- and ul-statements...

**Comment by BsNoSi** [comment]
For my homepage [https://nosi.de](nosi.de) ( if you want to see it "in action") I created a navigation, that feches depending on the current page sibling or child pages as links into a "sidebar" (→ "sub") that is part of the navigation. I use it for all pages from root down to 3rd level. »sub« is placed on the right as long there is enough space, otherwise it spreads over the whole page below main navigation if "hamburger menu" is active (screen width less 44 rem).

The relevant part of "header.html"

~~~.php
<nav id="navi1" class="navi">
<div class="navigation" role="navigation">
<?php foreach ($pages as $page): ?>
<div><a<?php echo $page->isActive() ? " class=\"active\" aria-current=\"page\"" : "" ?> href="<?php echo $page->getLocation(true) ?>"><?php echo $page->getHtml("titleNavigation") ?></a></div>
<?php endforeach ?>
</div>
<div class="sub">
<?php
$pages = $this->yellow->page->getChildren($showInvisible = false);
  if( ($this->yellow->page->getParent()) AND (count($pages)==0))  {  $pages = $this->yellow->page->getSiblings($showInvisible = false);   }
   if(count($pages))  {
	foreach($pages as $page) {				  
        $title = $page->get("titleNavigation");
        $curr = $page -> isActive() ? ' class="active" aria-current="page"' : '';
		echo '<div><a ' . $curr . 'href="' . $page->getLocation(true) . '">' . htmlspecialchars($title) . '</a></div>';
	 }
  } 
?>
</div>
</nav>
~~~

Some CSS manages formating in the content around the »sub« division:

~~~.css
.sub{min-width:10rem;max-width:30%;float:right;padding:0 0 0.2em 1em;margin:0 0 0 2em;background-color:#FFFFFF;border-left:0.1rem #d70 solid;}
.sub div{display:block;width:100%;padding-top:0.2rem;}
~~~
